collector.py

```diff
@@ -97,11 +120,21 @@ def collect(config_path, qcow, sps_version):
         virt_platform["hosts"][hostname] = state_obj.hardware_info(hostname)
         img = "%s-%s.img.qcow2" % (global_conf["hosts"][hostname]["profile"],
                                    sps_version)
+
+        if images_url and (img not in images_checksums):
+            img_checksum = img.replace("qcow2", "md5")
+            checksum = _get_checksum("%s/%s/%s" % (images_url, sps_version,
+                                                   img_checksum))
+            images_checksums[img] = checksum
+
         if qcow or \
            global_conf["hosts"][hostname]["profile"] == "install-server":
             if 'disks' not in virt_platform["hosts"][hostname]:
                 virt_platform["hosts"][hostname]["disks"] = [{'size': '15Gi'}]
             virt_platform["hosts"][hostname]["disks"][0]['image'] = img
+            if images_url:
```

Comment:
Not a big deal, but I think we can avoid this 3rd level of `if`. The code here is a bit cryptic.